modular/gater/request_context.go

```diff
@@ -237,59 +260,73 @@ func (r *RequestContext) verifyTaskSignature(taskMsgBytes []byte, taskSignature
 	return addr, nil
 }
 
-// verifyOffChainSignature used to verify off-chain-auth signature, return (address, nil) if check succeed
-func (r *RequestContext) verifyOffChainSignature(requestSignature string) (sdk.AccAddress, error) {
+// verifySignatureForGNFD1Eddsa used to verify off-chain-auth signature, return (address, nil) if check succeed
+func (r *RequestContext) verifySignatureForGNFD1Eddsa(requestSignature string) (sdk.AccAddress, error) {
 	var (
-		signedMsg *string
-		err       error
+		err error
 	)
-	signedMsg, sigString, err := parseSignedMsgAndSigFromRequest(requestSignature)
+	sigString, err := parseSignatureFromRequest(requestSignature)
 	if err != nil {
 		return nil, err
 	}
 
+	// check request integrity
+	realMsgToSign := commonhttp.GetMsgToSignInGNFD1Auth(r.request)
+
 	account := r.request.Header.Get(GnfdUserAddressHeader)
 	domain := r.request.Header.Get(GnfdOffChainAuthAppDomainHeader)
 	offChainSig := *sigString
-	realMsgToSign := *signedMsg
 
-	verifyOffChainSignatureResp, err := r.g.baseApp.GfSpClient().VerifyOffChainSignature(r.Context(), account, domain, offChainSig, realMsgToSign)
+	_, err = r.g.baseApp.GfSpClient().VerifyGNFD1EddsaSignature(r.Context(), account, domain, offChainSig, realMsgToSign)
 	if err != nil {
-		log.Errorf("failed to verify off chain signature", "error", err)
+		log.Errorf("failed to verify signature for GNFD1-Eddsa", "error", err)
 		return nil, err
-	}
-	if verifyOffChainSignatureResp {
+	} else {
 		userAddress, _ := sdk.AccAddressFromHexUnsafe(r.request.Header.Get(GnfdUserAddressHeader))
 		return userAddress, nil
-	} else {
+	}
+}
+
+// verifyOffChainSignatureFromPreSignedURL used to verify off-chain-auth signature, return (address, nil) if check succeed.  The auth information will be parsed from URL.
+func (r *RequestContext) verifyGNFD1EddsaSignatureFromPreSignedURL(authenticationStr string, account string, domain string) (sdk.AccAddress, error) {
+	var (
+		err error
+	)
+	sigString, err := parseSignatureFromRequest(authenticationStr)
+	if err != nil {
 		return nil, err
 	}
+
+	// check request integrity
+	realMsgToSign := commonhttp.GetMsgToSignInGNFD1AuthForPreSignedURL(r.request)
+
+	offChainSig := *sigString
+
+	_, err = r.g.baseApp.GfSpClient().VerifyGNFD1EddsaSignature(r.Context(), account, domain, offChainSig, realMsgToSign)
+	if err != nil {
+		log.Errorf("failed to verify off chain signature", "error", err)
+		return nil, err
+	} else {
+		userAddress, _ := sdk.AccAddressFromHexUnsafe(account)
+		return userAddress, nil
+	}
 }
 
-func parseSignedMsgAndSigFromRequest(requestSignature string) (*string, *string, error) {
+// parseSignatureFromRequest get sig for both ECDSA and EDDSA auth, it expects the auth string should look like "Signature=xxxxx".
+func parseSignatureFromRequest(requestSignature string) (*string, error) {
```

Comment:
Please dont use `*string`, but use `string` directly.  `string` is a pointer underlying.